pkg/cluster/clusterserviceprincipal.go

```diff
@@ -29,13 +29,12 @@ func (m *manager) createOrUpdateClusterServicePrincipalRBAC(ctx context.Context)
 		return err
 	}
 
-	// If we have Contributor RBAC role for Cluster SP on the resource group in question
 	// We are interested in Resource group scope only (inherited are returned too).
 	var toDelete []mgmtauthorization.RoleAssignment
 	var found bool
 	for _, assignment := range roleAssignments {
-		// Contributor assignments only!
-		if strings.EqualFold(*assignment.Scope, resourceGroupID) && strings.HasSuffix(strings.ToLower(*assignment.RoleDefinitionID), rbac.RoleContributor) {
+		if strings.EqualFold(*assignment.Scope, resourceGroupID) &&
+			!strings.HasSuffix(strings.ToLower(*assignment.RoleDefinitionID), strings.ToLower(rbac.RoleOwner)) /* should only matter in development */ {
 			if strings.EqualFold(*assignment.PrincipalID, clusterSPObjectID) {
```

Comment:
If im reading this right, this make that would be able to have only 1 CustomRole with RG scope for clusterSPObject? I just want to make sure mental picture is right

Reply:
Yes, at least for now